message/validation/alan/common_checks.go

```diff
@@ -0,0 +1,91 @@
+package msgvalidation
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/alan/types"
+	"golang.org/x/exp/slices"
+)
+
+// ownCommittee should be called only when WithOwnOperatorID is set
+func (mv *messageValidator) ownCommittee(committee []spectypes.OperatorID) bool {
+	return slices.ContainsFunc(committee, func(operatorID spectypes.OperatorID) bool {
+		return operatorID == mv.operatorDataStore.GetOperatorID()
+	})
+}
+
+func (mv *messageValidator) committeeRole(role spectypes.RunnerRole) bool {
+	return role == spectypes.RoleCommittee
+}
+
+func (mv *messageValidator) validateSlotTime(messageSlot phase0.Slot, role spectypes.RunnerRole, receivedAt time.Time) error {
+	if mv.earlyMessage(messageSlot, receivedAt) {
+		return ErrEarlyMessage
+	}
+
+	if lateness := mv.lateMessage(messageSlot, role, receivedAt); lateness > 0 {
+		e := ErrLateMessage
+		e.got = fmt.Sprintf("late by %v", lateness)
+		return e
+	}
+
+	return nil
+}
+
+func (mv *messageValidator) earlyMessage(slot phase0.Slot, receivedAt time.Time) bool {
+	return mv.netCfg.Beacon.GetSlotEndTime(mv.netCfg.Beacon.EstimatedSlotAtTime(receivedAt.Unix())).
+		Add(-clockErrorTolerance).Before(mv.netCfg.Beacon.GetSlotStartTime(slot))
+}
+
+func (mv *messageValidator) lateMessage(slot phase0.Slot, role spectypes.RunnerRole, receivedAt time.Time) time.Duration {
+	var ttl phase0.Slot
+	switch role {
+	case spectypes.RoleProposer, spectypes.RoleSyncCommitteeContribution:
+		ttl = 1 + lateSlotAllowance
```

Comment:
I guess it should be added ValidatorRegistration and VoluntaryExit. What do you think?

Reply:
@MatheusFranco99 but they don't have a consensus phase, so this code is unreachable for them

Reply:
Oh, I see. Thanks for the clarification